### ISSUE TITLE
Circular import in salt.utils.jinja

### ISSUE THREAD

**Issue by anlutro** [saltstack/salt#29960]
I was going to write some tests for something in `salt.utils.jinja` when I found out I can't import it due to a circular import:

```
$ python -c 'import tests.unit.utils.jinja_test'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "tests/unit/utils/jinja_test.py", line 13, in <module>
    from salt.utils.jinja import SerializerExtension
  File "salt/utils/jinja.py", line 27, in <module>
    import salt.fileclient
  File "salt/fileclient.py", line 27, in <module>
    import salt.utils.templates
  File "salt/utils/templates.py", line 28, in <module>
    from salt.utils.jinja import ensure_sequence_filter, show_full_context
ImportError: cannot import name ensure_sequence_filter
```

This is in 2015.8 HEAD.

**Comment by ssgward** [comment]
@anlutro - we'll look into this.

**Comment by rallytime** [comment]
@anlutro This should be fixed with #30043.
